R/SimpleReporter.R

```diff
@@ -36,8 +38,18 @@ NULL
 
 #' @rdname simple_reporter
 #' @export
-simple_reporter_ui <- function(id) {
+simple_reporter_ui <- function(
+    id,
+    modules = getOption("teal.reporter.simple_reporter_modules", c("add", "download", "load", "reset"))
```

Comment:
Same for the srv as well. We should consider changes in teal independently.
```suggestion
```

@donyunardi @kumamiao 
Problem is that each of teal_module in the app has
    - add button to add a current module's content to the Reporter
    - download button to download Reporter containing all modules' content
    - reset button to remove all modules' content from the Reporter
    - (new, introduced in this PR) button to load previously saved Reporter-files in order to restore Reporter

In the same time Previewer has load, download and reset. Problem for me is that we have load, download, reset in each module and in a previewer module. They are duplicated, it's literally the same button doing the same global-action. To solve this we might consider:

- replacing [teal.reporter::simple_report_previewer_ui](https://github.com/insightsengineering/teal.modules.general/blob/78b95ac65d9ba0e324419469ccedc4b1a2262156/R/tm_a_pca.R#L204) and [teal.reporter::simple_report_previewer_srv](https://github.com/insightsengineering/teal.modules.general/blob/78b95ac65d9ba0e324419469ccedc4b1a2262156/R/tm_a_pca.R#L1071) with just `add_card_button_ui` and `add_card_button_srv` to have teal_module tabs only adding reporter-cards.


![image](https://github.com/insightsengineering/teal.reporter/assets/6959016/bd97becb-2541-4d55-b822-949c6f2e82bf)